### ISSUE TITLE
LXR Lexer Error for Special Characters in Oracle Dialect Identifiers Cannot Be Ignored

### ISSUE THREAD

**Issue by revton** [sqlfluff/sqlfluff#7142]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I encountered an issue where SQLFluff's lexer fails to parse identifiers containing special characters (like 'ç',
  'ã', 'õ', etc.) in the Oracle dialect.
  

  This issue prevents proper linting of valid Oracle SQL that contains international characters in identifiers, which
  are valid in Oracle databases when properly configured with UTF-8 support

### Expected Behaviour

  Oracle databases support UTF-8 characters in identifiers, and SQLFluff should be able to parse identifiers with
  special characters like _reativação_ without throwing lexer errors.

### Observed Behaviour

  The lexer fails completely when encountering special characters in identifiers, resulting in LXR (lexer) and PRS
  (parser) violations that cannot be excluded via configuration.

  **Error:**                                                                                                                             
   - LXR: Unable to lex characters: 'ção'                                                                                            
   - PRS: Line 2, Position 19: Found unparsable section: 'ção'

### How to reproduce

**Test case:**
_sqlfluff_test/simple_special_char/query_lint.sql_
```sql
SELECT
    'Não' reativação
FROM
    TABLE1
```

  **Configuration used:**
_sqlfluff_test/simple_special_char/.sqlfluff_
```toml
[sqlfluff]
dialect = oracle
encoding = utf-8
exclude_rules = LT09, LT01, AL02, RF05, CP02, LT12
```
**Pytest**
```python
from pathlib import Path

import sqlfluff
from sqlfluff.core import FluffConfig

def test_simple_special_char():
    path = Path("sqlfluff_test/simple_special_char/")
    config = FluffConfig.from_path(path)
    with open(path / "query_lint.sql", encoding="utf-8") as sql_file:
        sql_query = sql_file.read()
    violations = sqlfluff.lint(sql_query, config=config)
    assert len(violations) == 0
```

Current output:
```shell
E       assert 2 == 0
E        +  where 2 = len([{'code': 'LXR', 'description': "Unable to lex characters: 'ção'", 'name': '', 'start_line_no': 2, ...}, {'code': 'PRS', 'description': "Line 2, Position 19: Found unparsable section: 'ção'", 'end_file_pos': 31, 'end_line_no': 2, ...}])
```

  **Environment:**
   - SQLFluff version: 3.0.4
   - Python version: 3.12.2
   - OS: Windows

### Dialect

oracle

### Version

sqlfluff, version 3.0.4

### Configuration

```toml
[sqlfluff]
dialect = oracle
encoding = utf-8
exclude_rules = LT09, LT01, AL02, RF05, CP02, LT12
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
